src/error.ts

```diff
@@ -200,6 +200,8 @@ export class MongoError extends Error {
  * @category Error
  */
 export class MongoServerError extends MongoError {
+  /** Raw error result document returned by server. */
+  errorResponse?: ErrorDescription;
```

Comment:
Should this be required?  We always assign it in the constructor

Reply:
Yep, just changed it.

Reply:
Ideally it should be. There were some type errors since we feed `MongoError` instances into `MongoServerError` fields in parts of the driver, so I just casted these instances as `MongoServerError` as a fix.

Reply:
Can you file a subtask to fix the type errors this uncovered (tackling it as follow-up is fine since it's a separate issue)?  I think it's a bug in the driver's types but we'll need to look to be sure - ideally we'd fix the type issue instead of just casting it away.

We have a number of instances of `MongoServerError` that are actually invoked with `MongoError`s, which worked before because `MongoError` was assignable to `MongoServerError`.

Reply:
Sure, here's the link [to the ticket]( https://jira.mongodb.org/browse/NODE-5999).